src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2531,6 +2627,32 @@ private void onDecodeTransfer(
                     {
                         flags = batchable(flags);
                     }
+                    if (amqpSection != null)
+                    {
+                        AmqpDescribedType type = amqpSection.kind();
+                        switch (type)
+                        {
+                        case DATA:
+                            bodyKind = AmqpBodyKind.DATA;
+                            break;
+                        case SEQUENCE:
+                            bodyKind = AmqpBodyKind.SEQUENCE;
+                            break;
+                        case VALUE:
+                            AmqpType valueType = amqpSection.amqpValue().kind();
+                            switch (valueType)
+                            {
+                            case STRING1:
+                            case STRING4:
+                                bodyKind = AmqpBodyKind.VALUE_STRING;
+                                break;
+                            case BINARY1:
+                            case BINARY4:
+                                bodyKind = AmqpBodyKind.VALUE_BINARY;
+                            }
+                            break;
+                        }
+                    }
```

Comment:
Recommend hoisting this to a `static` helper method, passing `AmqpSectionFW` returning `AmqpBodyKind`.